src/components/Loader.vue

```diff
@@ -0,0 +1,323 @@
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Watch, Emit, Vue } from 'vue-property-decorator';
+
+/**
+ * **Nimiq Loader Component**
+ *
+ * Props:
+ *
+ * **title** {string} The current title, dynamic for both loading and result states
+ *
+ * **status** {string} [optional] Currently doing this
+ *
+ * **state** {'loading'|'success'|'warning'|'error'} [optional, default 'loading']
+ *
+ * **lightBlue** {boolean} [optional, default false] Show light blue loading screen
+ *
+ * **message** {string} [optional] Message displayed for warning and error states
+ *
+ * **mainAction** {string} [optional] Text of main action button (button is hidden otherwise)
+ *
+ * **alternativeAction** {string} [optional] Text of alternative action link (link is hidden otherwise)
+ *
+ * Events:
+ *
+ * **@main-action**
+ *
+ * **@alternative-action**
+ *
+ * The `state` is also available as Loader.LOADING, Loader.SUCCESS, Loader.WARNING and Loader.ERROR.
+ */
+@Component
+export default class Loader extends Vue {
+    private static LOADING = 'loading';
+    private static SUCCESS = 'success';
+    private static WARNING = 'warning';
+    private static ERROR   = 'error';
+
+    @Prop({type: String, default: 'Improving the world'}) private title!: string;
+    @Prop({type: String, default: Loader.LOADING}) private state!: string;
+    @Prop(Boolean) private lightBlue?: boolean;
+    @Prop(String) private status?: string;
+    @Prop(String) private message?: string;
+    @Prop(String) private mainAction?: string;
+    @Prop(String) private alternativeAction?: string;
+
+    private currentStatus: string = '';
+    private nextStatus: string = '';
+    private isStatusTransitioning: boolean = false;
+
+    /**
+     * To enable a smooth transition of the non-transitionable background-image
+     * property, we instead place the new background above the old one and
+     * animate the top element's opacity. But because the color area has rounded
+     * corners, and the browser creates transparent pixels in the corner
+     * because of anti-aliasing, the blue background partly shines through the
+     * transparent corner pixels of the foreground. Thus we remove the background
+     * color after the transition is complete.
+     */
+    private showLoadingBackground: boolean = true;
+
+    private _loadingTitle?: string;
+
+    private get loadingTitle() {
+        return this._loadingTitle || (this._loadingTitle = this.title);
+    }
+
+    @Watch('state', {immediate: true})
+    private updateState(newState: string, oldState: string) {
+        // When the component is initialized with a state other than LOADING
+        if (!oldState && (newState !== Loader.LOADING)) {
+            this.showLoadingBackground = false;
+        }
+
+        // When the state changes later and animates
+        if (oldState && (newState !== Loader.LOADING)) {
+            setTimeout(() => this.showLoadingBackground = false, 1000);
+        }
```

Comment:
If the state changes back to `LOADING`, `showLoadingBackground` needs to be reset to `true` and the timeout cancelled.

Reply:
The component is not meant to be switched back, but I see your point.

Reply:
Resolved with 4e4d856.

Reply:
Not clearing the timeout though